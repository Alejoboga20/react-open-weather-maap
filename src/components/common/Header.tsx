import '../../styles/components/Header.css';

export interface HeaderProps {
  title: string;
  subtitle?: string;
}

export const Header = ({ title, subtitle = '' }: HeaderProps) => {
  return (
    <div className='header_container'>
      <h1 className='title'>{title}</h1>
      <h2 className='subtitle'>{subtitle}</h2>
      <hr />
    </div>
  );
};
