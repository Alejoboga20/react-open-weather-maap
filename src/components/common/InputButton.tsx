import { Coordinates } from '../../interfaces/Locations';
import '../../styles/components/InputButton.css';

export interface InputButtonProps {
  label: string;
  value: string;
  location: Coordinates;
  onClick: (event: any) => void;
  disabled: boolean;
}

export const InputButton = ({
  label,
  value,
  onClick,
  disabled,
}: InputButtonProps) => (
  <button
    value={value}
    onClick={onClick}
    disabled={disabled}
    className='inputButton'
  >
    {label}
  </button>
);
