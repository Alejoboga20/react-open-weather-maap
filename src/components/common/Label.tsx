import '../../styles/components/Label.css';

export interface LabelProps {
  labelContent: string;
}

export const Label = ({ labelContent }: LabelProps) => {
  return (
    <div className='label_container'>
      <label className='main_content'>{labelContent}</label>
    </div>
  );
};
