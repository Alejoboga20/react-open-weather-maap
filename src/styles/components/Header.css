.container {
  text-align: center;
}

.title {
  font-size: 600;
}

.subtitle {
  font-size: 400;
  font-weight: lighter;
}
