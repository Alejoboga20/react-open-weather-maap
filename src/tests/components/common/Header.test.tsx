import React from 'react';
import { shallow } from 'enzyme';
import { Header, HeaderProps } from '../../../components/common/Header';

const headerProps: HeaderProps = {
  title: 'Test Title',
  subtitle: 'Test Subtitle',
};

describe('Header Tests', () => {
  const wrapper = shallow(<Header {...headerProps} />);

  test('should match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have correct h1 and h2', () => {
    const title = wrapper.find('h1').text();
    const subtitle = wrapper.find('h2').text();

    expect({ title, subtitle }).toEqual(headerProps);
  });
});
