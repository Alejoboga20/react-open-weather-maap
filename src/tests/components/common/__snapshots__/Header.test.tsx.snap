// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests should match Snapshot 1`] = `
<div
  className="container"
>
  <h1
    className="title"
  >
    Test Title
  </h1>
  <h2
    className="subtitle"
  >
    Test Subtitle
  </h2>
  <hr />
</div>
`;
